pkg/dazzle/combiner_test.go

```diff
@@ -0,0 +1,185 @@
+	}{
+		{
+			name: "EnvVarCombineMergeUnique",
+			base: &ociv1.Image{
+				Config: ociv1.ImageConfig{
+					Env: []string{
+						"PATH=first:second:third:$PATH",
+					},
+				},
+			},
+			others: []*ociv1.Image{
+				{
+					Config: ociv1.ImageConfig{
+						Env: []string{
+							"PATH=fourth:fifth:$PATH",
+						},
+					},
+				},
+				{
+					Config: ociv1.ImageConfig{
+						Env: []string{
+							"PATH=sixth:sixth:$PATH",
+						},
+					},
+				},
+				{
+					Config: ociv1.ImageConfig{
+						Env: []string{
+							"PATH=eighth:seventh:eighth:$PATH",
+						},
+					},
+				},
+			},
+			vars: []EnvVarCombination{
+				{
+					Name:   "PATH",
+					Action: EnvVarCombineMergeUnique,
+				},
+			},
+			expect: []string{"PATH=first:second:third:fourth:fifth:sixth:seventh:eighth:$PATH"},
+		},
+		{
+			name: "EnvVarCombineMerge",
+			base: &ociv1.Image{
+				Config: ociv1.ImageConfig{
+					Env: []string{
+						"PATH=first:second:third:$PATH",
+					},
+				},
+			},
+			others: []*ociv1.Image{
+				{
+					Config: ociv1.ImageConfig{
+						Env: []string{
+							"PATH=fourth:fifth:$PATH",
+						},
+					},
+				},
+				{
+					Config: ociv1.ImageConfig{
+						Env: []string{
+							"PATH=sixth:sixth:$PATH",
+						},
+					},
+				},
+				{
+					Config: ociv1.ImageConfig{
+						Env: []string{
+							"PATH=eighth:seventh:eighth:$PATH",
+						},
+					},
+				},
+			},
+			vars: []EnvVarCombination{
+				{
+					Name:   "PATH",
+					Action: EnvVarCombineMerge,
+				},
+			},
+			expect: []string{"PATH=first:second:third:$PATH:fourth:fifth:$PATH:sixth:sixth:$PATH:eighth:seventh:eighth:$PATH"},
+		},
+		{
+			name: "EnvVarCombineUseLast",
+			base: &ociv1.Image{
+				Config: ociv1.ImageConfig{
+					Env: []string{
+						"PATH=first:second:third:$PATH",
+					},
+				},
+			},
+			others: []*ociv1.Image{
+				{
+					Config: ociv1.ImageConfig{
+						Env: []string{
+							"PATH=fourth:fifth:$PATH",
+						},
+					},
+				},
+			},
+			vars: []EnvVarCombination{
+				{
+					Name:   "PATH",
+					Action: EnvVarCombineUseLast,
+				},
+			},
+			expect: []string{"PATH=fourth:fifth:$PATH"},
+		},
+		{
+			name: "EnvVarCombineUseFirst",
+			base: &ociv1.Image{
+				Config: ociv1.ImageConfig{
+					Env: []string{
+						"PATH=first:second:third:$PATH",
+					},
+				},
+			},
+			others: []*ociv1.Image{
+				{
+					Config: ociv1.ImageConfig{
+						Env: []string{
+							"PATH=fourth:fifth:$PATH",
+						},
+					},
+				},
+			},
+			vars: []EnvVarCombination{
+				{
+					Name:   "PATH",
+					Action: EnvVarCombineUseFirst,
+				},
+			},
+			expect: []string{"PATH=first:second:third:$PATH"},
+		},
+	}
+	for _, test := range tests {
+		envs, err := mergeEnv(test.base, test.others, test.vars)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(envs) != len(test.expect) {
+			t.Fatal("unexpected length", len(envs))
+		}
+		for i, env := range envs {
+			if env != test.expect[i] {
+				t.Fatal("unexpected env", envs)
+			}
```

Comment:
Thank you very much for writing the test! 🚀 
By (still undocumented) convention, we use `go-cmp` for unit tests, e.g. here:
https://github.com/gitpod-io/dazzle/blob/949a695eb26b335f266839bfb2b10e60c3336a27/pkg/dazzle/project_test.go#L129-L131